packages/engine/lib/execution/src/task/shared_store.rs

```diff
@@ -41,17 +41,17 @@ pub struct PartialStateWriteProxy {
     pub state_proxy: StateWriteProxy,
 }
 
-/// Partial read access to the state pools.
+/// Partial read access to the agent and message pool.
```

Comment:
Tiny nit, sorry :D 

```suggestion
/// Partial read access to the agent and message pools.
```
or 
```suggestion
/// Partial read access to the agent pool and message pool.
```

(and on the other occurrences)

Reply:
I chose the latter to distinguish between singularity and plurality of pools.